src/os/bluestore/BlueStore.cc

```diff
@@ -7246,8 +7247,25 @@ int BlueStore::_do_alloc_write(
     CompressorRef c;
     bool compressed = false;
     if (wctx->compress &&
-	wi.blob_length > min_alloc_size &&
-	(c = compressor) != nullptr) {
+      wi.blob_length > min_alloc_size) {
+      c = select_option(
+	coll,
+	"compression_algorithm",
+	compressor,
+	[&]() {
+	  string val;
+	  if (coll->pool_opts.get(pool_opts_t::COMPRESSION_ALGORITHM, &val)) {
+	    CompressorRef cp = compressor;
+	    if (!cp || cp->get_type_name() != val) {
+	      cp = Compressor::create(cct, val);
+            }
+	    return boost::optional<CompressorRef>(cp);
+	  }
+	  return boost::optional<CompressorRef>();
+	}
+      );
```

Comment:
I think an easier helper would be a Collection method that's something like

</pre>
template<T>
T get_option(option, T default_value) {
   ...
}
</pre>
so that all these callers are simply 
<pre>
c = coll->get_option(pool_opts_t::COMPRESSION_ALGORITHM, compressor);
</pre>
or maybe a pool_opts_t method like
<pre>
template<typename T>
T get(int optid, T default) { ... }
</pre>
(ala python's dict get).